hoot-core/src/main/cpp/hoot/core/io/HootNetworkRequest.cpp

```diff
@@ -146,6 +174,17 @@ bool HootNetworkRequest::_networkRequest(QUrl url, const QMap<QNetworkRequest::K
   return true;
 }
 
+void HootNetworkRequest::_blockOnReply(QNetworkReply* reply)
+{
+  if (reply != NULL)
+  {
+    //  Qt code to force the use of QNetworkReply to be synchronous
+    QEventLoop loop;
+    QObject::connect(reply, SIGNAL(finished()), &loop, SLOT(quit()));
+    loop.exec();
+  }
+}
+
```

Comment:
Just a quick helper function to block the thread while the network request completes.